pkg/controller/vitesskeyspace/reconcile_topo.go

```diff
@@ -56,38 +55,21 @@ func (r *ReconcileVitessKeyspace) reconcileTopology(ctx context.Context, vtk *pl
 	defer ts.Close()
 
 	if *vtk.Spec.TopologyReconciliation.PruneShards {
-		result, err := r.pruneShards(ctx, vtk, ts)
-		resultBuilder.Merge(result, err)
-	}
-
-	return resultBuilder.Result()
-}
-
-func (r *ReconcileVitessKeyspace) pruneShards(ctx context.Context, vtk *planetscalev2.VitessKeyspace, ts *toposerver.Conn) (reconcile.Result, error) {
-	resultBuilder := &results.Builder{}
+		desiredShards := make(sets.String, len(vtk.Status.Shards))
+		for k := range vtk.Status.Shards {
+			desiredShards.Insert(k)
+		}
 
-	// Get list of shards in topo.
-	shardNames, err := ts.GetShardNames(ctx, vtk.Spec.Name)
-	if err != nil {
-		r.recorder.Eventf(vtk, corev1.EventTypeWarning, "TopoListFailed", "failed to list shards in topology: %v", err)
-		return resultBuilder.RequeueAfter(topoRequeueDelay)
+		result, err := vitesstopo.PruneShards(ctx, vitesstopo.PruneShardsParams{
```

Comment:
I know I didn't really contest this on the `PruneKeyspace` PR but what was ultimately the readability decision to have `PruneShards` take in its own custom type? Wouldn't it be cleaner in the main reconcile path just to pass in ctx, vtk, etc.?

Reply:
Per our offline discussion, I'll resolve this for now, with the idea that we table this discussion for later :smile_cat: